src/Stratis.Bitcoin/Base/Deployments/ThresholdConditionCache.cs

```diff
@@ -58,6 +59,55 @@ public ThresholdState[] GetStates(ChainedHeader pindexPrev)
             return array;
         }
 
+        public List<ThresholdStateModel> GetThresholdStateMetrics(ChainedHeader indexPrev, ThresholdState[] thresholdStates)
+        {
+            var thresholdStateModels = new List<ThresholdStateModel>();
+            ThresholdState[] array = new ThresholdState[this.consensus.BIP9Deployments.Length];
+
+            for (int deploymentIndex = 0; deploymentIndex < array.Length; deploymentIndex++)
+            {
+                if (this.consensus.BIP9Deployments[deploymentIndex] == null) continue;
+
+                DateTime? timeStart = this.consensus.BIP9Deployments[deploymentIndex]?.StartTime.Date;
+                DateTime? timeTimeout = this.consensus.BIP9Deployments[deploymentIndex]?.Timeout.Date;
+                int threshold = this.consensus.RuleChangeActivationThreshold;
+
+                int votes = 0;
```

Comment:
The votes calculated here are the votes counted in the most recent window. If the state is already `Active` then this vote count may be irrelevant and **would not necessarily equal the number of votes that led to activation**.

Reply:
Done